rocAL/source/pipeline/master_graph.cpp

```diff
@@ -1158,12 +1157,51 @@ TensorListVector* MasterGraph::create_label_reader(const char *source_path, Meta
     return &_metadata_output_tensor_list;
 }
 
-TensorListVector* MasterGraph::create_video_label_reader(const char *source_path, MetaDataReaderType reader_type, unsigned sequence_length, unsigned frame_step, unsigned frame_stride, bool file_list_frame_num) {
+TensorListVector* MasterGraph::create_webdataset_reader(
+    const char *source_path, const char *index_path,
+    std::vector<std::set<std::string>> extensions,
+    MetaDataReaderType reader_type, MissingComponentsBehaviour missing_component_behaviour) {
     if (_meta_data_reader)
         THROW("A metadata reader has already been created")
     if (_augmented_meta_data)
         THROW("Metadata can only have a single output")
 
+    bool generate_index = (index_path[0] == '\0') ? true : false;
+    if (generate_index)
+        std::cerr << "Index file is not provided, it may take some time to infer it from the tar file";
```

Comment:
do you want to continue the rest of the code after this error? why are we not using ERRLOG macro

Reply:
This statement is printed if the user doesn't pass the idx file for the webdataset tar file. We have added support to infer the file contents from the tar file if its not present.
We cannot use ERR macro here as this is not an error but a warning that it will take some time for inferring file contents. There are INFO and WRN macros which can be used here but we are not enabling them in CMakelists. Only ERR and THROW macros are available for use